code/game/atoms_movable.dm

```diff
@@ -83,6 +83,10 @@
 	/// The degree of pressure protection that mobs in list/contents have from the external environment, between 0 and 1
 	var/contents_pressure_protection = 0
 
+	///some movables need a concept of "real loc" thats separate from the loc var because they get nullspaced. no handling for this should exist at the
+	///movable level, subtypes that use it should do all handling themselves.
+	var/turf/associated_loc
```

Comment:
Is there a way to mark an object as using this system? or do they need to manage it manually

Reply:
its managed by the object. the way everything currently works is that if associated_loc is defined then the object gets nullspaced (whether or not its currently nullspaced)

Reply:
Looking through here you have a few different implementations of the same pattern, each with slight differences
I wonder if it wouldn't be better to make using nullspaced contents a flag, rather then implementing it individually
Though I suppose this is supposed to be hard to extend